### ISSUE TITLE
Could not parse the Youtube page for URL http://youtube.com/watch?v=

### ISSUE THREAD

**Issue by hazemkut1993** [flagbug/YoutubeExtractor#329]
i'm using YoutubeExtractor in my vb.net project to download the youtube

but i'm getting this erorr 

> 
    Could not parse the Youtube page for URL http://youtube.com/watch?v=YQHsXMglC9A This may be due to a change of the Youtube page structure. Please report this bug at www.github.com/flagbug/YoutubeExtractor/issues

this image for the erorr

https://i.imgur.com/g8bUrlk.png

my code is 

```
Private Sub Button5_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button5.Click
        Dim vInfos As IEnumerable(Of VideoInfo)
        vInfos = DownloadUrlResolver.GetDownloadUrls(TextBox1.Text, False)


        ComboBox1.Items.Clear()

        Dim listDL As New List(Of String)
        Dim listExt As New List(Of String)

        Dim blnFlag As Boolean = True
        For Each vi As VideoInfo In vInfos
            If blnFlag = True Then
                strDefTitle = vi.Title
                blnFlag = False
            End If
            Dim strQuality As String
            strQuality = "Resoultion: " & vi.Resolution & " Format: " & vi.VideoExtension
            ComboBox1.Items.Add(strQuality)

            'If the video has a decrypted signature, decipher it
            If vi.RequiresDecryption Then
                DownloadUrlResolver.DecryptDownloadUrl(vi)
            End If

            listDL.Add(vi.DownloadUrl)
            listExt.Add(vi.VideoExtension)

        Next
        strArrayLinks = listDL.ToArray
        strArrayExts = listExt.ToArray

    End Sub
```

**Comment by janakiram-a** [comment]
Hi hazemkut1993,
I am also facing the same issue right now.
Please let me know if you get any solution.
Thanks

**Comment by janluijk** [comment]
I am working with c# and get the same error: 

YoutubeExtractor.YoutubeParseException: 'Could not parse the Youtube page for URL http://youtube.com/watch?v=MwjsO6aniig
This may be due to a change of the Youtube page structure.
Please report this bug at www.github.com/flagbug/YoutubeExtractor/issues'

**Comment by johnalp** [comment]
I faced the same issue, here how I fixed it:

Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:

private static string GetHtml5PlayerVersion(JObject json)
    {
        var regex = new Regex(@"player_ias-(.+?).js");

        string js = json["assets"]["js"].ToString();

        Match match = regex.Match(js);
        if (match.Success) return match.Result("$1");

        regex = new Regex(@"player-(.+?).js");
        
        return regex.Match(js).Result("$1");
    }

You will then need to rebuild the YoutubeExtractor.dll and it should be good!

**Comment by janakiram-a** [comment]
> I faced the same issue, here how I fixed it:
> 
> Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:
> 
> private static string GetHtml5PlayerVersion(JObject json)
> {
> var regex = new Regex(@"player_ias-(.+?).js");
> 
> ```
>     string js = json["assets"]["js"].ToString();
> 
>     Match match = regex.Match(js);
>     if (match.Success) return match.Result("$1");
> 
>     regex = new Regex(@"player-(.+?).js");
>     
>     return regex.Match(js).Result("$1");
> }
> ```
> You will then need to rebuild the YoutubeExtractor.dll and it should be good!

Hi johnalp,
Your suggestion really helped me a lot.
Many thanks and good wishes to you.

**Comment by SoftwareMagicIT** [comment]
Today: could not decipher signature issue

**Comment by janakiram-a** [comment]
> Today: could not decipher signature issue

Try another youtube url to download. I had the same issue with my old test url. I tried new one and it worked.

**Comment by anilgit90** [comment]
Please look at pull request #333 , I hope it will solve your problems.

**Comment by kumkumbum001** [comment]
> > I faced the same issue, here how I fixed it:
> > Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:
> > private static string GetHtml5PlayerVersion(JObject json)
> > {
> > var regex = new Regex(@"player_ias-(.+?).js");
> > ```
> >     string js = json["assets"]["js"].ToString();
> > 
> >     Match match = regex.Match(js);
> >     if (match.Success) return match.Result("$1");
> > 
> >     regex = new Regex(@"player-(.+?).js");
> >     
> >     return regex.Match(js).Result("$1");
> > }
> > ```
> > You will then need to rebuild the YoutubeExtractor.dll and it should be good!
> 
> Hi johnalp,
> Your suggestion really helped me a lot.
> Many thanks and good wishes to you.


Can you help me? I don't know how to fix it! Hope you help me!
My FB: https://www.facebook.com/phong.truong.1990

**Comment by kumkumbum001** [comment]
> I faced the same issue, here how I fixed it:
> 
> Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:
> 
> private static string GetHtml5PlayerVersion(JObject json)
> {
> var regex = new Regex(@"player_ias-(.+?).js");
> 
> ```
>     string js = json["assets"]["js"].ToString();
> 
>     Match match = regex.Match(js);
>     if (match.Success) return match.Result("$1");
> 
>     regex = new Regex(@"player-(.+?).js");
>     
>     return regex.Match(js).Result("$1");
> }
> ```
> You will then need to rebuild the YoutubeExtractor.dll and it should be good!


Give me the file please YoutubeExtractor.dll can you fix the error?
My mail: fungzhangteemo@gmail.com

**Comment by kumkumbum001** [comment]
> > I faced the same issue, here how I fixed it:
> > Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:
> > private static string GetHtml5PlayerVersion(JObject json)
> > {
> > var regex = new Regex(@"player_ias-(.+?).js");
> > ```
> >     string js = json["assets"]["js"].ToString();
> > 
> >     Match match = regex.Match(js);
> >     if (match.Success) return match.Result("$1");
> > 
> >     regex = new Regex(@"player-(.+?).js");
> >     
> >     return regex.Match(js).Result("$1");
> > }
> > ```
> > You will then need to rebuild the YoutubeExtractor.dll and it should be good!
> 
> Hi johnalp,
> Your suggestion really helped me a lot.
> Many thanks and good wishes to you.

Give me the file please YoutubeExtractor.dll can you fix the error?
My mail: fungzhangteemo@gmail.com

**Comment by SoftwareMagicIT** [comment]
Hi kumkumbum001, you can stop to SPAM this thread? If you don't know how to solve, because info is very easy to this, we can't help you. Simply you need to edit DownloadUrlResolver.cs file in source code project, replace GetHtml5PlayerVersion() with piece of code suggested, recompile and use. If you don't know to made this, we can't help you!

**Comment by kumkumbum001** [comment]
I don't know how to open the file  DownloadUrlResolver.cs  . Can you help
me open that file?

Vào Th 6, 4 thg 1, 2019 vào lúc 00:13 Software Magic <
notifications@github.com> đã viết:

**Comment by kumkumbum001** [comment]
I don't want to spam, but I'm really having trouble! Can you send me
YoutubeExtractor-Fix?

Vào Th 6, 4 thg 1, 2019 vào lúc 09:53 Trương Hoài Phong <
fungzhangteemo@gmail.com> đã viết:

**Comment by SoftwareMagicIT** [comment]
No! On github the source code is shared to be improved by developers. If you do not know how to do it, you need to use stable release. If the stable release does not work, you have to wait for the owner to release a stable version. Alternatively you install Visual Studio Community, you download the source and you fill it in yourself, after the appropriate modifications. I do not want to be responsible for providing a dll compiled by me. If you repeat same message or similar message, you generate SPAM! If you need you can use edit function of comment

**Comment by rafadante** [comment]
> I faced the same issue, here how I fixed it:
> 
> Replace GetHtml5PlayerVersion() in DownloadUrlResolver.cs with this:
> 
> private static string GetHtml5PlayerVersion(JObject json)
> {
> var regex = new Regex(@"player_ias-(.+?).js");
> 
> ```
>     string js = json["assets"]["js"].ToString();
> 
>     Match match = regex.Match(js);
>     if (match.Success) return match.Result("$1");
> 
>     regex = new Regex(@"player-(.+?).js");
>     
>     return regex.Match(js).Result("$1");
> }
> ```
> You will then need to rebuild the YoutubeExtractor.dll and it should be good!

Thanks, you really helped a lot
